lib/importers/daily_calls/tp/retriever.rb

```diff
@@ -0,0 +1,31 @@
+require 'mail_retriever'
+
+module Importers
+  module DailyCalls
+    module TP
+      class Retriever
+        class MissingProcessingBlock < StandardError; end
+
+        SEARCH_KEYS = ['SUBJECT', 'TP Daily Call Data'].freeze
+        FILE_NAME_REGEXP = /Daily Data File.*\.xlsx/
+
+        def initialize(config:, mail_retriever: MailRetriever)
+          @config = config
+          @mail_retriever = mail_retriever
+        end
+
+        def process_emails
+          raise MissingProcessingBlock unless block_given?
+
+          retriever = @mail_retriever.new(config: @config)
+          emails = retriever.search(search_keys: SEARCH_KEYS, file_name_regexp: FILE_NAME_REGEXP)
+
+          emails.each do |email|
+            yield email
+            retriever.archive(uid: email.uid)
```

Comment:
What happens if this call fails? Does the same attachment get processed twice or is it idempotent?